ovs/lib/vdisk.py

```diff
@@ -526,26 +527,37 @@ def create_new(diskname, size, storagedriver_guid):
 
     @staticmethod
     @celery.task(name='ovs.vdisk.create_volume')
-    def create_volume(location, size):
+    def create_volume(location, size, storagerouter_guid=None):
         """
         Create a volume using filesystem calls
         Calls "truncate" to create sparse raw file
-        TODO: use volumedriver API
-        TODO: model VDisk() and return guid
 
         @param location: location, filename
         @param size: size of volume, GB
+        @param: storagerouter_guid: use ssh client to create file on remote storagerouter
         @return None
         """
         logger.info('Creating volume {0} of {1} GB'.format(location, size))
-        if os.path.exists(location):
-            raise RuntimeError('File already exists at %s' % location)
+        if storagerouter_guid is None:
```

Comment:
I'd merge the 2 paths to prevent code duplication
client = SSHClient(storagerouter) if storagerouter is not None else SSHClient('127.0.0.1')

Then you can execute all calls using the client and the SSHClient will use check_output itself when executed locally